app/src/main/java/xyz/block/gosling/features/agent/Agent.kt

```diff
@@ -129,12 +143,7 @@ class Agent : Service() {
         try {
             isCancelled = false
 
-            val availableIntents = IntentScanner.getAvailableIntents(
-                context,
-                GoslingAccessibilityService.getInstance()
-            )
-            val installedApps = IntentAppKinds.groupIntentsByCategory(availableIntents)
-
+            val installedApps = getInstalledApps(context)
```

Comment:
Ideally we would refresh this when the app becomes active again. Otherwise if you install an app and then ask gosling something it won't know about it until gosling gets restarted (which since it has a background task won't happen easily)

Reply:
yeah - could be same with crawling for other tools/MCPs - would app active event be the go - that means everytime you go back to gosling it does it, which isn't that different to the experience now? (or maybe are better events)

Reply:
Yeah - we should this on start up and then when we become active again, kick off a thread to refresh ideally